### PR TITLE
Dotnet tests

### PR DESCRIPTION
**Pull request by AndyHeap-NeoTech**

### CODE REVIEW

File: tests/stub/routing/test_routing_v4x3.py [hunk 1]
```diff
@@ -415,7 +415,7 @@ def test_should_write_successfully_on_writer_using_tx_run(self):
     def test_should_write_successfully_on_writer_using_tx_function(self):
         # TODO remove this block once all languages work
         if get_driver_name() in ['dotnet']:
-            self.skipTest("crashes the backend")
+           self.skipTest("crashes the backend")
```
Comment (robsdedude) [inline tests/stub/routing/test_routing_v4x3.py]:
```suggestion
            self.skipTest("crashes the backend")
```

File: tests/stub/routing/test_routing_v4x3.py [hunk 2]
```diff
@@ -152,7 +152,7 @@ def test_should_send_system_bookmark_with_route(self):
     def test_should_read_successfully_from_reader_using_tx_function(self):
         # TODO remove this block once all languages work
         if get_driver_name() in ['dotnet']:
-            self.skipTest("crashes the backend")
+           self.skipTest("crashes the backend")
```
Comment (robsdedude) [inline tests/stub/routing/test_routing_v4x3.py]:
```suggestion
            self.skipTest("crashes the backend")
```

File: tests/stub/routing/test_routing_v4x3.py [hunk 3]
```diff
@@ -1478,7 +1470,7 @@ def work(tx):
     def test_should_successfully_read_from_readable_router_using_tx_function(self):
         # TODO remove this block once all languages work
         if get_driver_name() in ['dotnet']:
-            self.skipTest("needs ROUTE bookmark list support")
+            self.skipTest("Test failing for some reason")
```
Comment (robsdedude) [inline tests/stub/routing/test_routing_v4x3.py:1473]: Alternatively, you could consider moving the skip into the driver's TestKit backend.

**Review by robsdedude — CHANGES_REQUESTED** [review]

File: tests/stub/routing/test_routing_v4x3.py (hunk 3 above)
Comment (AndyHeap-NeoTech) [reply tests/stub/routing/test_routing_v4x3.py:1473]: That is the plan. I'm going to go through and start moving all the 'dotnet' if statements into the blacklist.

File: tests/stub/routing/test_routing_v4x3.py [hunk 4]
```diff
@@ -1217,11 +1212,8 @@ def work(tx):
         #       that `try_count == 2`. When doing so be aware that drivers could
         #       do round robin, e.g. Java.
 
-    def _should_retry_write_tx_until_success_on_error(
+    def test_should_retry_write_tx_until_success_on_error(
```
Comment (AndyHeap-NeoTech) [inline tests/stub/routing/test_routing_v4x3.py]:
```suggestion
    def _should_retry_write_tx_until_success_on_error(
```

Comment (AndyHeap-NeoTech) [inline tests/stub/routing/test_routing_v4x3.py]:
```suggestion
    def _should_retry_write_tx_until_success_on_error(
```

**Pull request merged**